From afd2fe90807de46476f1a50e84cdea5245b0e902 Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Wed, 27 Nov 2013 16:59:49 +0100
Subject: [PATCH 11/41] extlinux: Add test suite

Strongly inspired aby GRUB test suite. Note that this does not pass currently,
since the extlinux default line gets eaten (rhbz#964178). A fix will follow.

Signed-off-by: Lubomir Rintel <lkundrak@v3.sk>
---
 test.sh                          | 58 +++++++++++++++++++++++++++++++++++++++-
 test/extlinux.1                  | 17 ++++++++++++
 test/extlinux.2                  | 21 +++++++++++++++
 test/extlinux.3                  | 15 +++++++++++
 test/extlinux.4                  | 32 ++++++++++++++++++++++
 test/results/add/extlinux1.1     | 20 ++++++++++++++
 test/results/add/extlinux1.2     | 20 ++++++++++++++
 test/results/add/extlinux1.3     | 21 +++++++++++++++
 test/results/add/extlinux1.4     | 21 +++++++++++++++
 test/results/add/extlinux2.1     | 21 +++++++++++++++
 test/results/args/extlinux1.1    | 20 ++++++++++++++
 test/results/args/extlinux1.2    | 20 ++++++++++++++
 test/results/default/extlinux1.1 | 19 +++++++++++++
 test/results/default/extlinux1.2 | 19 +++++++++++++
 test/results/default/extlinux3.1 | 16 +++++++++++
 test/results/default/extlinux3.2 | 15 +++++++++++
 test/results/remove/extlinux3.1  | 10 +++++++
 test/results/remove/extlinux4.1  | 27 +++++++++++++++++++
 test/results/updargs/extlinux1.1 | 17 ++++++++++++
 test/results/updargs/extlinux1.2 | 18 +++++++++++++
 test/results/updargs/extlinux3.1 | 15 +++++++++++
 test/results/updargs/extlinux3.2 | 15 +++++++++++
 test/results/updargs/extlinux3.4 | 15 +++++++++++
 test/results/updargs/extlinux3.7 | 13 +++++++++
 test/results/updargs/extlinux4.2 | 32 ++++++++++++++++++++++
 test/results/updargs/extlinux4.3 | 32 ++++++++++++++++++++++
 test/results/updargs/extlinux4.4 | 32 ++++++++++++++++++++++
 test/results/updargs/extlinux4.5 | 33 +++++++++++++++++++++++
 28 files changed, 613 insertions(+), 1 deletion(-)
 create mode 100644 test/extlinux.1
 create mode 100644 test/extlinux.2
 create mode 100644 test/extlinux.3
 create mode 100644 test/extlinux.4
 create mode 100644 test/results/add/extlinux1.1
 create mode 100644 test/results/add/extlinux1.2
 create mode 100644 test/results/add/extlinux1.3
 create mode 100644 test/results/add/extlinux1.4
 create mode 100644 test/results/add/extlinux2.1
 create mode 100644 test/results/args/extlinux1.1
 create mode 100644 test/results/args/extlinux1.2
 create mode 100644 test/results/default/extlinux1.1
 create mode 100644 test/results/default/extlinux1.2
 create mode 100644 test/results/default/extlinux3.1
 create mode 100644 test/results/default/extlinux3.2
 create mode 100644 test/results/remove/extlinux3.1
 create mode 100644 test/results/remove/extlinux4.1
 create mode 100644 test/results/updargs/extlinux1.1
 create mode 100644 test/results/updargs/extlinux1.2
 create mode 100644 test/results/updargs/extlinux3.1
 create mode 100644 test/results/updargs/extlinux3.2
 create mode 100644 test/results/updargs/extlinux3.4
 create mode 100644 test/results/updargs/extlinux3.7
 create mode 100644 test/results/updargs/extlinux4.2
 create mode 100644 test/results/updargs/extlinux4.3
 create mode 100644 test/results/updargs/extlinux4.4
 create mode 100644 test/results/updargs/extlinux4.5

diff --git a/test.sh b/test.sh
index 1d28fb1..734b755 100755
--- a/test.sh
+++ b/test.sh
@@ -158,7 +158,7 @@ commandTest() {
 
 # generate convenience functions
 for b in $(./grubby --help | \
-	sed -n 's/^.*--\([^ ]*\) *configure \1 bootloader$/\1/p'); do
+	sed -n 's/^.*--\([^ ]*\) *configure \1 bootloader.*/\1/p'); do
     eval "${b}Test() { [[ \"$b\" == \$opt_bootloader ]] && oneTest --$b \"\$@\"; }"
     eval "${b}DisplayTest() { [[ \"$b\" == \$opt_bootloader ]] && oneDisplayTest --$b \"\$@\"; }"
 done
@@ -316,12 +316,24 @@ grubTest grub.5 fallback/g5.3 --remove-kernel=/boot/vmlinuz-2.4.7-2.5 \
     --boot-filesystem=/ --copy-default --add-kernel=/boot/new-kernel \
     --title="Some_Title"
 
+testing="Extlinux default directive"
+extlinuxTest extlinux.1 default/extlinux1.1 --boot-filesystem=/boot --add-kernel /boot/new-kernel --title Some_Title
+extlinuxTest extlinux.1 default/extlinux1.2 --boot-filesystem=/boot --add-kernel /boot/new-kernel --title Some_Title --make-default
+extlinuxTest extlinux.3 default/extlinux3.1 --boot-filesystem=/boot --set-default=/boot/vmlinuz-3.12.0-2.fc21.i686
+extlinuxTest extlinux.3 default/extlinux3.2 --boot-filesystem=/boot --set-default=/boot/vmlinuz-3.12.0-2.fc21.i686+PAE
+
 testing="GRUB new kernel argument handling"
 grubTest grub.1 args/g1.1 --boot-filesystem=/boot \
     --add-kernel=/boot/foo --title=some_title --args="1234" --copy-default
 grubTest grub.1 args/g1.2 --boot-filesystem=/boot \
     --add-kernel=/boot/foo --title=some_title --args="1234" 
 
+testing="Extlinux new kernel argument handling"
+extlinuxTest extlinux.1 args/extlinux1.1 --boot-filesystem=/boot \
+    --add-kernel=/boot/foo --title=some_title --args="1234" --copy-default
+extlinuxTest extlinux.1 args/extlinux1.2 --boot-filesystem=/boot \
+    --add-kernel=/boot/foo --title=some_title --args="1234"
+
 testing="GRUB remove kernel"
 grubTest grub.7 remove/g7.1 --boot-filesystem=/ \
     --remove-kernel=/boot/vmlinuz-2.4.7-2.5
@@ -339,6 +351,12 @@ testing="Z/IPL remove kernel"
 ziplTest zipl.1 remove/z1.1 --remove-kernel=/boot/vmlinuz-2.4.9-38
 ziplTest zipl.1 remove/z1.2 --remove-kernel=DEFAULT
 
+testing="Extlinux remove kernel"
+extlinuxTest extlinux.4 remove/extlinux4.1 --boot-filesystem=/ \
+    --remove-kernel=/boot/vmlinuz-3.11.7-301.fc20.i686
+extlinuxTest extlinux.3 remove/extlinux3.1 --boot-filesystem=/ \
+    --remove-kernel=DEFAULT
+
 testing="GRUB update kernel argument handling"
 grubTest grub.1 updargs/g1.1 --update-kernel=DEFAULT --args="root=/dev/hda1"
 grubTest grub.1 updargs/g1.2 --update-kernel=DEFAULT \
@@ -391,6 +409,30 @@ liloTest lilo.3 updargs/l3.1 --update-kernel=/boot/vmlinuz-2.4.18-4 \
 liloTest lilo.3 updargs/l3.2 --update-kernel=ALL \
     --remove-args="single" --args "root=/dev/hda2"
 
+testing="Extlinux update kernel argument handling"
+extlinuxTest extlinux.1 updargs/extlinux1.1 --update-kernel=DEFAULT --args="root=/dev/hda1"
+extlinuxTest extlinux.1 updargs/extlinux1.2 --update-kernel=DEFAULT \
+    --args="root=/dev/hda1 hda=ide-scsi root=/dev/hda2"
+extlinuxTest extlinux.3 updargs/extlinux3.1 --update-kernel=DEFAULT --args "hdd=notide-scsi"
+extlinuxTest extlinux.3 updargs/extlinux3.2 --update-kernel=DEFAULT \
+    --args "hdd=notide-scsi root=/dev/hdd1"
+extlinuxTest extlinux.3 updargs/extlinux3.2 --update-kernel=DEFAULT \
+    --args "root=/dev/hdd1 hdd=notide-scsi"
+extlinuxTest extlinux.3 updargs/extlinux3.4 --update-kernel=ALL --remove-args="hdd"
+extlinuxTest extlinux.3 updargs/extlinux3.4 --update-kernel=ALL --remove-args="hdd=ide-scsi"
+extlinuxTest extlinux.3 updargs/extlinux3.4 --update-kernel=ALL --remove-args="hdd=foobar"
+extlinuxTest extlinux.3 updargs/extlinux3.7 --update-kernel=ALL \
+    --remove-args="hdd root ro"
+extlinuxTest extlinux.4 updargs/extlinux4.2 --boot-filesystem=/    \
+    --update-kernel=ALL --args "hde=ide-scsi"
+extlinuxTest extlinux.4 updargs/extlinux4.3 --boot-filesystem=/    \
+    --update-kernel=DEFAULT --args "hde=ide-scsi"
+extlinuxTest extlinux.4 updargs/extlinux4.4 --boot-filesystem=/    \
+    --update-kernel=/vmlinuz-3.12.0-2.fc21.i686 \
+    --args "ro root=LABEL=/ console=tty0 console=ttyS1,9600n81 single"
+extlinuxTest extlinux.4 updargs/extlinux4.5 --boot-filesystem=/    \
+    --update-kernel=ALL --args "root=/dev/hda2"
+
 testing="LILO add kernel"
 liloTest lilo.4 add/l4.1 --add-kernel=/boot/new-kernel.img --title="title" \
     --copy-default --boot-filesystem=/boot
@@ -577,6 +619,20 @@ testing="Z/IPL add kernel"
 ziplTest zipl.1 add/z1.1 --add-kernel=/boot/new-kernel.img --title test
 ziplTest zipl.1 add/z1.2 --add-kernel=/boot/new-kernel.img --title test --copy-default
 
+testing="Extlinux add kernel"
+extlinuxTest extlinux.1 add/extlinux1.1 --add-kernel=/boot/new-kernel.img --title='title' \
+    --initrd=/boot/new-initrd --boot-filesystem=/
+extlinuxTest extlinux.1 add/extlinux1.2 --add-kernel=/boot/new-kernel.img --title='title' \
+    --initrd=/boot/new-initrd --boot-filesystem=/boot
+extlinuxTest extlinux.1 add/extlinux1.3 --add-kernel=/boot/new-kernel.img --title='title' \
+    --initrd=/boot/new-initrd --boot-filesystem=/ --copy-default
+extlinuxTest extlinux.1 add/extlinux1.4 --add-kernel=/boot/new-kernel.img --title='title' \
+    --initrd=/boot/new-initrd --boot-filesystem=/boot --copy-default
+extlinuxTest extlinux.2 add/extlinux2.1 --add-kernel=/boot/vmlinuz-3.12.0-2.fc21.i686	    \
+    --initrd=/boot/initrd-3.12.0-2.fc21.i686-new.img --boot-filesystem=/boot --copy-default \
+    --title="Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)"					    \
+    --remove-kernel="TITLE=Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)"
+
 testing="LILO long titles"
 liloTest lilo.1 longtitle/l1.1 --add-kernel=/boot/new-kernel.img \
     --title="linux-longtitle" --copy-default --boot-filesystem=/boot 
diff --git a/test/extlinux.1 b/test/extlinux.1
new file mode 100644
index 0000000..85c9272
--- /dev/null
+++ b/test/extlinux.1
@@ -0,0 +1,17 @@
+# extlinux.conf generated by anaconda
+
+ui menu.c32
+
+menu autoboot Welcome to Fedora. Automatic boot in # second{,s}. Press a key for options.
+menu title Fedora Boot Options.
+menu hidden
+
+timeout 50
+totaltimeout 9000
+
+default Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+append ro root=/dev/sda1
+initrd /initrd-3.12.0-2.fc21.i686.img
diff --git a/test/extlinux.2 b/test/extlinux.2
new file mode 100644
index 0000000..eae4aa7
--- /dev/null
+++ b/test/extlinux.2
@@ -0,0 +1,21 @@
+# extlinux.conf generated by anaconda
+
+ui menu.c32
+
+menu autoboot Welcome to Fedora. Automatic boot in # second{,s}. Press a key for options.
+menu title Fedora Boot Options.
+menu hidden
+
+timeout 50
+totaltimeout 9000
+
+default Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+append ro root=/dev/hda5 hdd=ide-scsi
+initrd /initrd-3.12.0-2.fc21.i686.img
+
+label DOS
+com32 chain.c32
+append hd0 1
diff --git a/test/extlinux.3 b/test/extlinux.3
new file mode 100644
index 0000000..63f97bf
--- /dev/null
+++ b/test/extlinux.3
@@ -0,0 +1,15 @@
+prompt 0
+
+label Fedora (3.12.0-2.fc21.i686+PAE) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686+PAE
+append ro root=/dev/sda1
+initrd /initrd-3.12.0-2.fc21.i686+PAE.img
+
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+append ro root=/dev/sda1
+initrd /initrd-3.12.0-2.fc21.i686.img
+
+label DOS
+com32 chain.c32
+append hd0 1
diff --git a/test/extlinux.4 b/test/extlinux.4
new file mode 100644
index 0000000..14b795c
--- /dev/null
+++ b/test/extlinux.4
@@ -0,0 +1,32 @@
+ui menu.c32
+
+menu hidden
+timeout 50
+totaltimeout 9000
+
+prompt 10
+default Fedora (3.13.0-0.rc0.git5.1.fc21.i686) 20 (Heisenbug)
+
+label Fedora (3.11.7-300.fc20.i686) 20 (Heisenbug)
+kernel /boot/vmlinuz-3.11.7-300.fc20.i686
+append ro root=/dev/hda6
+initrd /boot/initrd-3.11.7-300.fc20.i686.img
+
+label Fedora (3.11.7-301.fc20.i686) 20 (Heisenbug)
+kernel /boot/vmlinuz-3.11.7-301.fc20.i686
+append ro root=/dev/hda6
+initrd /boot/initrd-3.11.7-301.fc20.i686.img
+
+label Fedora (3.13.0-0.rc0.git5.1.fc21.i686) 20 (Heisenbug)
+kernel /boot/vmlinuz-3.13.0-0.rc0.git5.1.fc21.i686
+append ro root=/dev/hda6
+initrd /boot/initrd-3.13.0-0.rc0.git5.1.fc21.i686.img
+
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+append ro root=/dev/sda1 console=tty0 console=ttyS1,9600n81
+initrd /initrd-3.12.0-2.fc21.i686.img
+
+label DOS
+com32 chain.c32
+append hd0 1
diff --git a/test/results/add/extlinux1.1 b/test/results/add/extlinux1.1
new file mode 100644
index 0000000..f81a7ed
--- /dev/null
+++ b/test/results/add/extlinux1.1
@@ -0,0 +1,20 @@
+# extlinux.conf generated by anaconda
+
+ui menu.c32
+
+menu autoboot Welcome to Fedora. Automatic boot in # second{,s}. Press a key for options.
+menu title Fedora Boot Options.
+menu hidden
+
+timeout 50
+totaltimeout 9000
+
+default Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+
+label title
+kernel /boot/new-kernel.img
+initrd /boot/new-initrd
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+append ro root=/dev/sda1
+initrd /initrd-3.12.0-2.fc21.i686.img
diff --git a/test/results/add/extlinux1.2 b/test/results/add/extlinux1.2
new file mode 100644
index 0000000..71e0437
--- /dev/null
+++ b/test/results/add/extlinux1.2
@@ -0,0 +1,20 @@
+# extlinux.conf generated by anaconda
+
+ui menu.c32
+
+menu autoboot Welcome to Fedora. Automatic boot in # second{,s}. Press a key for options.
+menu title Fedora Boot Options.
+menu hidden
+
+timeout 50
+totaltimeout 9000
+
+default Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+
+label title
+kernel /new-kernel.img
+initrd /new-initrd
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+append ro root=/dev/sda1
+initrd /initrd-3.12.0-2.fc21.i686.img
diff --git a/test/results/add/extlinux1.3 b/test/results/add/extlinux1.3
new file mode 100644
index 0000000..9634025
--- /dev/null
+++ b/test/results/add/extlinux1.3
@@ -0,0 +1,21 @@
+# extlinux.conf generated by anaconda
+
+ui menu.c32
+
+menu autoboot Welcome to Fedora. Automatic boot in # second{,s}. Press a key for options.
+menu title Fedora Boot Options.
+menu hidden
+
+timeout 50
+totaltimeout 9000
+
+default Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+
+label title
+kernel /boot/new-kernel.img
+append ro root=/dev/sda1
+initrd /boot/new-initrd
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+append ro root=/dev/sda1
+initrd /initrd-3.12.0-2.fc21.i686.img
diff --git a/test/results/add/extlinux1.4 b/test/results/add/extlinux1.4
new file mode 100644
index 0000000..aa1ce63
--- /dev/null
+++ b/test/results/add/extlinux1.4
@@ -0,0 +1,21 @@
+# extlinux.conf generated by anaconda
+
+ui menu.c32
+
+menu autoboot Welcome to Fedora. Automatic boot in # second{,s}. Press a key for options.
+menu title Fedora Boot Options.
+menu hidden
+
+timeout 50
+totaltimeout 9000
+
+default Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+
+label title
+kernel /new-kernel.img
+append ro root=/dev/sda1
+initrd /new-initrd
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+append ro root=/dev/sda1
+initrd /initrd-3.12.0-2.fc21.i686.img
diff --git a/test/results/add/extlinux2.1 b/test/results/add/extlinux2.1
new file mode 100644
index 0000000..13a4179
--- /dev/null
+++ b/test/results/add/extlinux2.1
@@ -0,0 +1,21 @@
+# extlinux.conf generated by anaconda
+
+ui menu.c32
+
+menu autoboot Welcome to Fedora. Automatic boot in # second{,s}. Press a key for options.
+menu title Fedora Boot Options.
+menu hidden
+
+timeout 50
+totaltimeout 9000
+
+default Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+append ro root=/dev/hda5 hdd=ide-scsi
+initrd /initrd-3.12.0-2.fc21.i686-new.img
+
+label DOS
+com32 chain.c32
+append hd0 1
diff --git a/test/results/args/extlinux1.1 b/test/results/args/extlinux1.1
new file mode 100644
index 0000000..2d8c800
--- /dev/null
+++ b/test/results/args/extlinux1.1
@@ -0,0 +1,20 @@
+# extlinux.conf generated by anaconda
+
+ui menu.c32
+
+menu autoboot Welcome to Fedora. Automatic boot in # second{,s}. Press a key for options.
+menu title Fedora Boot Options.
+menu hidden
+
+timeout 50
+totaltimeout 9000
+
+default Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+
+label some_title
+kernel /foo
+append ro root=/dev/sda1 1234
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+append ro root=/dev/sda1
+initrd /initrd-3.12.0-2.fc21.i686.img
diff --git a/test/results/args/extlinux1.2 b/test/results/args/extlinux1.2
new file mode 100644
index 0000000..41cc324
--- /dev/null
+++ b/test/results/args/extlinux1.2
@@ -0,0 +1,20 @@
+# extlinux.conf generated by anaconda
+
+ui menu.c32
+
+menu autoboot Welcome to Fedora. Automatic boot in # second{,s}. Press a key for options.
+menu title Fedora Boot Options.
+menu hidden
+
+timeout 50
+totaltimeout 9000
+
+default Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+
+label some_title
+kernel /foo
+append 1234
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+append ro root=/dev/sda1
+initrd /initrd-3.12.0-2.fc21.i686.img
diff --git a/test/results/default/extlinux1.1 b/test/results/default/extlinux1.1
new file mode 100644
index 0000000..5d7f73d
--- /dev/null
+++ b/test/results/default/extlinux1.1
@@ -0,0 +1,19 @@
+# extlinux.conf generated by anaconda
+
+ui menu.c32
+
+menu autoboot Welcome to Fedora. Automatic boot in # second{,s}. Press a key for options.
+menu title Fedora Boot Options.
+menu hidden
+
+timeout 50
+totaltimeout 9000
+
+default Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+
+label Some_Title
+kernel /new-kernel
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+append ro root=/dev/sda1
+initrd /initrd-3.12.0-2.fc21.i686.img
diff --git a/test/results/default/extlinux1.2 b/test/results/default/extlinux1.2
new file mode 100644
index 0000000..e98ff12
--- /dev/null
+++ b/test/results/default/extlinux1.2
@@ -0,0 +1,19 @@
+# extlinux.conf generated by anaconda
+
+ui menu.c32
+
+menu autoboot Welcome to Fedora. Automatic boot in # second{,s}. Press a key for options.
+menu title Fedora Boot Options.
+menu hidden
+
+timeout 50
+totaltimeout 9000
+
+default Some_Title
+
+label Some_Title
+kernel /new-kernel
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+append ro root=/dev/sda1
+initrd /initrd-3.12.0-2.fc21.i686.img
diff --git a/test/results/default/extlinux3.1 b/test/results/default/extlinux3.1
new file mode 100644
index 0000000..5a5495d
--- /dev/null
+++ b/test/results/default/extlinux3.1
@@ -0,0 +1,16 @@
+prompt 0
+
+default=Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+label Fedora (3.12.0-2.fc21.i686+PAE) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686+PAE
+append ro root=/dev/sda1
+initrd /initrd-3.12.0-2.fc21.i686+PAE.img
+
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+append ro root=/dev/sda1
+initrd /initrd-3.12.0-2.fc21.i686.img
+
+label DOS
+com32 chain.c32
+append hd0 1
diff --git a/test/results/default/extlinux3.2 b/test/results/default/extlinux3.2
new file mode 100644
index 0000000..63f97bf
--- /dev/null
+++ b/test/results/default/extlinux3.2
@@ -0,0 +1,15 @@
+prompt 0
+
+label Fedora (3.12.0-2.fc21.i686+PAE) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686+PAE
+append ro root=/dev/sda1
+initrd /initrd-3.12.0-2.fc21.i686+PAE.img
+
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+append ro root=/dev/sda1
+initrd /initrd-3.12.0-2.fc21.i686.img
+
+label DOS
+com32 chain.c32
+append hd0 1
diff --git a/test/results/remove/extlinux3.1 b/test/results/remove/extlinux3.1
new file mode 100644
index 0000000..12b4ef5
--- /dev/null
+++ b/test/results/remove/extlinux3.1
@@ -0,0 +1,10 @@
+prompt 0
+
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+append ro root=/dev/sda1
+initrd /initrd-3.12.0-2.fc21.i686.img
+
+label DOS
+com32 chain.c32
+append hd0 1
diff --git a/test/results/remove/extlinux4.1 b/test/results/remove/extlinux4.1
new file mode 100644
index 0000000..5e1824e
--- /dev/null
+++ b/test/results/remove/extlinux4.1
@@ -0,0 +1,27 @@
+ui menu.c32
+
+menu hidden
+timeout 50
+totaltimeout 9000
+
+prompt 10
+default Fedora (3.13.0-0.rc0.git5.1.fc21.i686) 20 (Heisenbug)
+
+label Fedora (3.11.7-300.fc20.i686) 20 (Heisenbug)
+kernel /boot/vmlinuz-3.11.7-300.fc20.i686
+append ro root=/dev/hda6
+initrd /boot/initrd-3.11.7-300.fc20.i686.img
+
+label Fedora (3.13.0-0.rc0.git5.1.fc21.i686) 20 (Heisenbug)
+kernel /boot/vmlinuz-3.13.0-0.rc0.git5.1.fc21.i686
+append ro root=/dev/hda6
+initrd /boot/initrd-3.13.0-0.rc0.git5.1.fc21.i686.img
+
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+append ro root=/dev/sda1 console=tty0 console=ttyS1,9600n81
+initrd /initrd-3.12.0-2.fc21.i686.img
+
+label DOS
+com32 chain.c32
+append hd0 1
diff --git a/test/results/updargs/extlinux1.1 b/test/results/updargs/extlinux1.1
new file mode 100644
index 0000000..1e2a709
--- /dev/null
+++ b/test/results/updargs/extlinux1.1
@@ -0,0 +1,17 @@
+# extlinux.conf generated by anaconda
+
+ui menu.c32
+
+menu autoboot Welcome to Fedora. Automatic boot in # second{,s}. Press a key for options.
+menu title Fedora Boot Options.
+menu hidden
+
+timeout 50
+totaltimeout 9000
+
+default Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+append ro root=/dev/hda1
+initrd /initrd-3.12.0-2.fc21.i686.img
diff --git a/test/results/updargs/extlinux1.2 b/test/results/updargs/extlinux1.2
new file mode 100644
index 0000000..764655a
--- /dev/null
+++ b/test/results/updargs/extlinux1.2
@@ -0,0 +1,18 @@
+# extlinux.conf generated by anaconda
+
+ui menu.c32
+
+menu autoboot Welcome to Fedora. Automatic boot in # second{,s}. Press a key for options.
+menu title Fedora Boot Options.
+menu hidden
+
+timeout 50
+totaltimeout 9000
+
+default Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+append ro root=/dev/hda1 hda=ide-scsi
+initrd /initrd-3.12.0-2.fc21.i686.img
+root /dev/hda2
diff --git a/test/results/updargs/extlinux3.1 b/test/results/updargs/extlinux3.1
new file mode 100644
index 0000000..ef9fa86
--- /dev/null
+++ b/test/results/updargs/extlinux3.1
@@ -0,0 +1,15 @@
+prompt 0
+
+label Fedora (3.12.0-2.fc21.i686+PAE) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686+PAE
+append ro root=/dev/sda1 hdd=notide-scsi
+initrd /initrd-3.12.0-2.fc21.i686+PAE.img
+
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+append ro root=/dev/sda1
+initrd /initrd-3.12.0-2.fc21.i686.img
+
+label DOS
+com32 chain.c32
+append hd0 1
diff --git a/test/results/updargs/extlinux3.2 b/test/results/updargs/extlinux3.2
new file mode 100644
index 0000000..c85b6bf
--- /dev/null
+++ b/test/results/updargs/extlinux3.2
@@ -0,0 +1,15 @@
+prompt 0
+
+label Fedora (3.12.0-2.fc21.i686+PAE) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686+PAE
+append ro root=/dev/hdd1 hdd=notide-scsi
+initrd /initrd-3.12.0-2.fc21.i686+PAE.img
+
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+append ro root=/dev/sda1
+initrd /initrd-3.12.0-2.fc21.i686.img
+
+label DOS
+com32 chain.c32
+append hd0 1
diff --git a/test/results/updargs/extlinux3.4 b/test/results/updargs/extlinux3.4
new file mode 100644
index 0000000..63f97bf
--- /dev/null
+++ b/test/results/updargs/extlinux3.4
@@ -0,0 +1,15 @@
+prompt 0
+
+label Fedora (3.12.0-2.fc21.i686+PAE) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686+PAE
+append ro root=/dev/sda1
+initrd /initrd-3.12.0-2.fc21.i686+PAE.img
+
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+append ro root=/dev/sda1
+initrd /initrd-3.12.0-2.fc21.i686.img
+
+label DOS
+com32 chain.c32
+append hd0 1
diff --git a/test/results/updargs/extlinux3.7 b/test/results/updargs/extlinux3.7
new file mode 100644
index 0000000..b542ad3
--- /dev/null
+++ b/test/results/updargs/extlinux3.7
@@ -0,0 +1,13 @@
+prompt 0
+
+label Fedora (3.12.0-2.fc21.i686+PAE) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686+PAE
+initrd /initrd-3.12.0-2.fc21.i686+PAE.img
+
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+initrd /initrd-3.12.0-2.fc21.i686.img
+
+label DOS
+com32 chain.c32
+append hd0 1
diff --git a/test/results/updargs/extlinux4.2 b/test/results/updargs/extlinux4.2
new file mode 100644
index 0000000..9bf16b7
--- /dev/null
+++ b/test/results/updargs/extlinux4.2
@@ -0,0 +1,32 @@
+ui menu.c32
+
+menu hidden
+timeout 50
+totaltimeout 9000
+
+prompt 10
+default Fedora (3.13.0-0.rc0.git5.1.fc21.i686) 20 (Heisenbug)
+
+label Fedora (3.11.7-300.fc20.i686) 20 (Heisenbug)
+kernel /boot/vmlinuz-3.11.7-300.fc20.i686
+append ro root=/dev/hda6 hde=ide-scsi
+initrd /boot/initrd-3.11.7-300.fc20.i686.img
+
+label Fedora (3.11.7-301.fc20.i686) 20 (Heisenbug)
+kernel /boot/vmlinuz-3.11.7-301.fc20.i686
+append ro root=/dev/hda6 hde=ide-scsi
+initrd /boot/initrd-3.11.7-301.fc20.i686.img
+
+label Fedora (3.13.0-0.rc0.git5.1.fc21.i686) 20 (Heisenbug)
+kernel /boot/vmlinuz-3.13.0-0.rc0.git5.1.fc21.i686
+append ro root=/dev/hda6 hde=ide-scsi
+initrd /boot/initrd-3.13.0-0.rc0.git5.1.fc21.i686.img
+
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+append ro root=/dev/sda1 console=tty0 console=ttyS1,9600n81 hde=ide-scsi
+initrd /initrd-3.12.0-2.fc21.i686.img
+
+label DOS
+com32 chain.c32
+append hd0 1 hde=ide-scsi
diff --git a/test/results/updargs/extlinux4.3 b/test/results/updargs/extlinux4.3
new file mode 100644
index 0000000..218711d
--- /dev/null
+++ b/test/results/updargs/extlinux4.3
@@ -0,0 +1,32 @@
+ui menu.c32
+
+menu hidden
+timeout 50
+totaltimeout 9000
+
+prompt 10
+default Fedora (3.13.0-0.rc0.git5.1.fc21.i686) 20 (Heisenbug)
+
+label Fedora (3.11.7-300.fc20.i686) 20 (Heisenbug)
+kernel /boot/vmlinuz-3.11.7-300.fc20.i686
+append ro root=/dev/hda6
+initrd /boot/initrd-3.11.7-300.fc20.i686.img
+
+label Fedora (3.11.7-301.fc20.i686) 20 (Heisenbug)
+kernel /boot/vmlinuz-3.11.7-301.fc20.i686
+append ro root=/dev/hda6
+initrd /boot/initrd-3.11.7-301.fc20.i686.img
+
+label Fedora (3.13.0-0.rc0.git5.1.fc21.i686) 20 (Heisenbug)
+kernel /boot/vmlinuz-3.13.0-0.rc0.git5.1.fc21.i686
+append ro root=/dev/hda6 hde=ide-scsi
+initrd /boot/initrd-3.13.0-0.rc0.git5.1.fc21.i686.img
+
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+append ro root=/dev/sda1 console=tty0 console=ttyS1,9600n81
+initrd /initrd-3.12.0-2.fc21.i686.img
+
+label DOS
+com32 chain.c32
+append hd0 1
diff --git a/test/results/updargs/extlinux4.4 b/test/results/updargs/extlinux4.4
new file mode 100644
index 0000000..dc298fb
--- /dev/null
+++ b/test/results/updargs/extlinux4.4
@@ -0,0 +1,32 @@
+ui menu.c32
+
+menu hidden
+timeout 50
+totaltimeout 9000
+
+prompt 10
+default Fedora (3.13.0-0.rc0.git5.1.fc21.i686) 20 (Heisenbug)
+
+label Fedora (3.11.7-300.fc20.i686) 20 (Heisenbug)
+kernel /boot/vmlinuz-3.11.7-300.fc20.i686
+append ro root=/dev/hda6
+initrd /boot/initrd-3.11.7-300.fc20.i686.img
+
+label Fedora (3.11.7-301.fc20.i686) 20 (Heisenbug)
+kernel /boot/vmlinuz-3.11.7-301.fc20.i686
+append ro root=/dev/hda6
+initrd /boot/initrd-3.11.7-301.fc20.i686.img
+
+label Fedora (3.13.0-0.rc0.git5.1.fc21.i686) 20 (Heisenbug)
+kernel /boot/vmlinuz-3.13.0-0.rc0.git5.1.fc21.i686
+append ro root=/dev/hda6
+initrd /boot/initrd-3.13.0-0.rc0.git5.1.fc21.i686.img
+
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+append ro root=LABEL=/ console=tty0 console=ttyS1,9600n81 single
+initrd /initrd-3.12.0-2.fc21.i686.img
+
+label DOS
+com32 chain.c32
+append hd0 1
diff --git a/test/results/updargs/extlinux4.5 b/test/results/updargs/extlinux4.5
new file mode 100644
index 0000000..6418a48
--- /dev/null
+++ b/test/results/updargs/extlinux4.5
@@ -0,0 +1,33 @@
+ui menu.c32
+
+menu hidden
+timeout 50
+totaltimeout 9000
+
+prompt 10
+default Fedora (3.13.0-0.rc0.git5.1.fc21.i686) 20 (Heisenbug)
+
+label Fedora (3.11.7-300.fc20.i686) 20 (Heisenbug)
+kernel /boot/vmlinuz-3.11.7-300.fc20.i686
+append ro root=/dev/hda2
+initrd /boot/initrd-3.11.7-300.fc20.i686.img
+
+label Fedora (3.11.7-301.fc20.i686) 20 (Heisenbug)
+kernel /boot/vmlinuz-3.11.7-301.fc20.i686
+append ro root=/dev/hda2
+initrd /boot/initrd-3.11.7-301.fc20.i686.img
+
+label Fedora (3.13.0-0.rc0.git5.1.fc21.i686) 20 (Heisenbug)
+kernel /boot/vmlinuz-3.13.0-0.rc0.git5.1.fc21.i686
+append ro root=/dev/hda2
+initrd /boot/initrd-3.13.0-0.rc0.git5.1.fc21.i686.img
+
+label Fedora (3.12.0-2.fc21.i686) 20 (Heisenbug)
+kernel /vmlinuz-3.12.0-2.fc21.i686
+append ro root=/dev/hda2 console=tty0 console=ttyS1,9600n81
+initrd /initrd-3.12.0-2.fc21.i686.img
+
+label DOS
+com32 chain.c32
+append hd0 1
+root /dev/hda2
-- 
2.4.3

